core/src/main/java/dev/vml/es/acm/core/code/Executor.java

```diff
@@ -109,9 +115,18 @@ private ImmediateExecution executeImmediately(ExecutionContext context) {
                 return execution.end(ExecutionStatus.SUCCEEDED);
             }
 
-            statuses.put(context.getId(), ExecutionStatus.RUNNING);
-            contentScript.run();
-            return execution.end(ExecutionStatus.SUCCEEDED);
+            try {
```

Comment:
clue of change is to protect only RUNNING phase (critical section) ; things before like checking canRun => condition.unlocked() are not in critical section